### PR TITLE
108 add support list parsing

### LINKED ISSUE #108: Add support for parsing list commands
We need to support parsing of flags for list command as well, so that the correct type of command `ListApplicantCommand` or `ListInterviewCommand` can be returned.

### PR DESCRIPTION
**Pull request by goalfix**
I created support for list parsing of appropriate types. Now, users can add a flag to the `list` command, such as `list -i`, to tell which data type they want to list.

Closes #108

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ListCommandParser.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+import seedu.address.logic.commands.position.ListPositionCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+
+
+
+public class ListCommandParser implements Parser<ListCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the AddApplicantCommand,
```
Comment (YiHern-Lee) [inline src/main/java/seedu/address/logic/parser/ListCommandParser.java:19]: I think you missed out on editing the javadoc here.

**Review by YiHern-Lee — CHANGES_REQUESTED** [review]
Please change the javadoc before merging. Thank you!

File: src/main/java/seedu/address/logic/parser/ListCommandParser.java (hunk 1 above)
Comment (likeabowx) [inline src/main/java/seedu/address/logic/parser/ListCommandParser.java:19]: This JavaDocs comment seems to be about `AddApplicantCommand`, do change it for this command accordingly.

**Review by likeabowx — CHANGES_REQUESTED** [review]
LGTM, but need to update the JavaDocs first.

File: src/main/java/seedu/address/logic/parser/ListCommandParser.java (hunk 1 above)
Comment (goalfix) [reply src/main/java/seedu/address/logic/parser/ListCommandParser.java]: Thanks for pointing this out. I have changed the javadoc as you suggested.

**Review by likeabowx — APPROVED** [review]
LGTM now :)

**Review by YiHern-Lee — APPROVED** [review]
LGTM!

**Pull request merged**